src/ign_TEST.cc

```diff
@@ -96,3 +97,33 @@ TEST_F(CmdLine, IGN_UTILS_TEST_ENABLED_ONLY_ON_LINUX(list))
   EXPECT_TRUE(common::exists(common::joinPaths(this->kFakeHome, ".ignition",
       "gui")));
 }
+
+//////////////////////////////////////////////////
+/// \brief Check --help message and bash completion script for consistent flags
+TEST(ignTest, GuiHelpVsCompletionFlags)
```

Comment:
The only concern I have with the way this test is currently structured is that it relies on developers remembering to update it whenever they add a new flag. And at this point, they may as well remember to update the bash completion as well.

Ideally the test wouldn't have the options hardcoded, so that whenever a new option is added, the test automatically detects that it's missing from bash completion.

Reply:
I'll add that the tests themselves are not foolproof, especially for the single letter flags. Strings like `-a` could match many things in a bash script file.

To automate the tests, I guess it could be possible to add a function to the bash scripts returning the word list, and the test can source the bash script and then call that function. It probably needs to be a separate function than the existing one that's used by the actual bash completion, so the word list might need to be a global variable.

Reply:
[6f7291b](https://github.com/gazebosim/gz-gui/pull/392/commits/6f7291bf3b06f9c71f5aa3eff7fc3c919b69209d) changes test so that it's no longer hardcoded. Flags are now in a variable in the bash script, and a new bash script function returns them. The test calls this function.